fe-library-app/src/components/AddModifyForm/AddModifyBookForm.tsx

```diff
@@ -0,0 +1,288 @@
+    {
+      getBookById(props.id)
+        .then((response) => {
+          setNewBookTitle(response.data.Title)
+          setNewBookDisc(response.data.Description)
+          setNewBookIsbn(response.data.ISBN)
+          setNewBookQuantity(response.data.Quantity)
+          setNewBookPublishDate(new Intl.DateTimeFormat('en-CA').format(new Date(response.data.PublishDate)))
+          setSelectedAuthors(response.data.Authors.map((author) => {
+            return {
+              Id: author.Id,
+              FirstName: author.Firstname,
+              LastName: author.Lastname
+            }
+          }))
+          setNewBookCoverShow('data:image/png;base64,' + response.data.Cover)
+        }).catch(() => alert('Something went wrong'))
+    }
+    getAuthors()
+      .then((response) => {
+        setAuthorList(response.data)
+      })
+      .catch(() => alert('Error with getting authors'))
+  }, [ props.id ])
+
+  useEffect(() => {
+    getAuthorList()
+  }, [ getAuthorList ])
+
+  const handleSelectChange = (selection: MultiValue<GetAuthorResponse>) => {
+    setSelectedAuthors(selection as GetAuthorResponse[])
+  }
+
+  const handleCoverChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const files = event.currentTarget.files
+    const reader = new FileReader()
+    if (files !== null) {
+      reader.readAsDataURL(files[0])
+      setNewBookCover(files[0])
+      reader.onloadend = () => {
+        if(reader.result)
+          setNewBookCoverShow(reader.result as string)
+      }
+    }
+  }
+
+  const handleTitleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setNewBookTitle(event.currentTarget.value)
+  }
+
+  const handleDescChange = (event: ChangeEvent<HTMLTextAreaElement>) => {
+    setNewBookDisc(event.currentTarget.value)
+  }
+
+  const handleIsbnChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setNewBookIsbn(event.currentTarget.value)
+  }
+
+  const handleQuantityChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setNewBookQuantity(event.currentTarget.value !== '' ? parseInt(event.currentTarget.value) : 0)
+  }
+
+  const handlePublishDateChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setNewBookPublishDate(event.currentTarget.value)
+  }
+
+  const handleTitleBlur = () => {
+    if (newBookTitle.trim().length === 0){
+      setInputError((inputError) => {return{ ...inputError, titleError: true }})
+    }
+  }
+
+  const handleIsbnBlur = () => {
+    const pattern = /^(?=(?:\D*\d){10}(?:(?:\D*\d){3})?$)[\d-]+$/
+    if (newBookIsbn.trim().length === 0)
+    {
+      setInputError((inputError) => {return{ ...inputError, isbnErrorEmpty: true }})
+      setIsbnErrorMessage('Isbn is required')
+    }
+    else if (pattern.test(newBookIsbn) === false)
+    {
+      setInputError((inputError) => {return{ ...inputError, isbnErrorFormat: true }})
+      setIsbnErrorMessage('Bad format of Isbn')
+    }
+  }
+
+  const handleQuantityBlur = () => {
+    if (newBookQuantity <= 0)
```

Comment:
You can add min/max on input so quantity will never be less than 0

Reply:
In that case I don't need Error checker for that property right?

Reply:
In that case you will only have to check if it is 0 and not less than 0, this works fine of course, but there is no need to allow user to go lower than 0 for quantity from UI at all